### PR TITLE
docs(gateway-protocol): align doc for form deletion

### LINKED ISSUE #14858: Add support to Java client for Form deletion
**Description**

Blocked by #14272

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

No much to do for java client as the most part are already implemented.

## Related issues

closes #14858 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: gateway-protocol/src/main/proto/gateway.proto [hunk 1]
```diff
@@ -613,7 +613,7 @@ message ModifyProcessInstanceResponse {
 
 message DeleteResourceRequest {
   // The key of the resource that should be deleted. This can either be the key
-  // of a process definition, or the key of a decision requirements definition.
+  // of a process definition, the key of a decision requirements definition ot the key of a form.
```
Comment (berkaycanbc) [inline gateway-protocol/src/main/proto/gateway.proto:616]:
```suggestion
  // of a process definition, the key of a decision requirements definition or the key of a form.
```

**Review by berkaycanbc — CHANGES_REQUESTED** [review]
Thanks @nicpuppa! Just added a suggestion for a small typo. 👍 

One addition, we might need to update this in the Camunda 8 Zeebe gRPC API docs as well.

**Comment by nicpuppa** [comment]
> One addition, we might need to update this in the Camunda 8 Zeebe gRPC API docs as well.

There is an issue for this right ? All the form documentation on camunda 8 docs need to be done

**Comment by berkaycanbc** [comment]
> > One addition, we might need to update this in the Camunda 8 Zeebe gRPC API docs as well.
> 
> There is an issue for this right ? All the form documentation on camunda 8 docs need to be done

Exactly. Even I created it . 😅 👍

**Review by berkaycanbc — APPROVED** [review]

**Comment by nicpuppa** [comment]
bors merge

**Comment by nicpuppa** [comment]
bors retry

**Pull request merged**